### PR TITLE
Don’t add error when changing checksum to `:no_check`.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

---

Fixes the CI error here: https://github.com/Homebrew/homebrew-cask/pull/51795

### CODE REVIEW

**Comment by reitermarkus** [comment]
Should changing from `:no_check` still be an error?

**Comment by claui** [comment]
> Should changing from `:no_check` still be an error?

I think it didn’t really matter until https://github.com/Homebrew/homebrew-cask/pull/51795 but we should definitely allow it from now on.

File: Library/Homebrew/cask/audit.rb [hunk 1]
```diff
@@ -133,6 +133,8 @@ def check_required_stanzas
     end
 
     def check_version_and_checksum
+      return if cask.sha256 == :no_check
```
Comment (claui) [inline Library/Homebrew/cask/audit.rb]: Do we really want to allow `:no_check` for any constellation? Or just for documented cases of known-volatile SHAs, such as https://github.com/Homebrew/homebrew-cask/pull/51795?

Comment (reitermarkus) [reply Library/Homebrew/cask/audit.rb]: I think we *already* allow it. This is just to skip the check when the checksum changes.

Comment (claui) [reply Library/Homebrew/cask/audit.rb]: > I think we already allow it.

Indeed. TIL!

**Pull request merged**